### DISCUSSION TITLE
checkbox with wire:model and wire:click event propagation...

### DISCUSSION THREAD

**Discussion by dgillier** [livewire/livewire#4654] [1. Help] [answered]
Hello 

I'm trying to use Livewire on a table with the following row features:
- A checkbox in the row allowing its selection
- A click on the row itself redirecting to another page
- A click to a certain column field in the row opening a modal

Modal is working fine, as the click event on the field is not propagating itself to the row thanks to "wire:click.stop".

But my problem is the click on the checkbox, the click event on the checkbox is propagated to the row and is redirecting to another page...
Of course I tried the click modifiers on wire.model ("stop", " prevent", "self"...) but it's not working...
Any idea on how to solve this ?

Thanks , Denis

```
<tr class="cursor-pointer" wire:click="showRow({{ $video->id }})">
      <td>
          <input type="checkbox" value="{{ $video->id }}" wire:model="selected">
      </td>

       <td>
           ...
       </td>

       <td class ='cursor-pointer' wire:click.stop="$emit('openModal','modal-bookings',{{ json_encode(["event" => $event->id]) 
           }})">
      </td>
</tr>
```

**Comment by silnarm** [comment, marked as answer]
You could put a wire:click.stop (with no 'value') on the checkbox,
`<input type="checkbox" value="{{ $video->id }}" wire:model="selected" wire:click.stop="" />`

Perhaps not the most elegant/clear looking, but does work.

I see an unrelated issue in you code though, it is binding every checkbox to the same property (safely assuming the snippet you posted is in a loop). Because of the value attrib (the presence of which alerted me to a potential problem) the value for $selected will be the id of the last thing checked (or false if the last thing was unchecked), you have no way of knowing which other things were selected, and as these are checkboxes the truthyness of the property bound is the only things that matters, so if you check one, they will **all** "light up".

If you didn't intend to support multiple selection, just change the input type to 'radio' and give them a name (the same on all) and I think everything else you have should work.

If you did want multi-selection, your selected property needs to be an array, and bound accordingly.

If you only want to support selections on the current result set (collection that is paged or filtered somehow) then you can index that array by the index in the collection, whenever you change the collection you then need to reset the selection array: `$this->selected = array_fill(0, count($this->things), 0);`
The binding then looks like:
```
@foreach ($things as $index => $thing)
              <tr wire:click="showRow({{$thing->id}})">
                <td><input type="checkbox" value="1" wire:model="selected.{{$index}}" wire:click.stop="" /></td>
```

If you want to support selections that span pages/result-sets, then you need to key the selected array by something unique (model pk most likely), and not clear it when the result set is updated. Beware the UX issues here though.

**Reply by dgillier** [reply]
@silnarm Thanks a lot !
Works great! Would not have came to my mind to try that ;)

And thanks also for the selection code :)
